lib/spack/spack/database.py

```diff
@@ -103,29 +125,47 @@ class InstallRecord(object):
     actually remove from the database until a spec has no installed
     dependents left.
 
+    Args:
+        spec (Spec): spec tracked by the install record
+        path (str): path where the spec has been installed
+        installed (bool): whether or not the spec is currently installed
+        ref_count (int): number of specs that depend on this one
```

Comment:
I actually agree with @healther here.  Don't use `**kwargs` for everything; [use explicit arguments](https://stackoverflow.com/a/30882832/9122).  I realize that allows them to be positional, but we can enforce that at the callsite instead of in the function.  We could even add a style check for it.  I think the benefits of good documentation and method signatures outweigh the use of `**kwargs`, and I had started switching away from `**kwargs` where possible.

In google pydoc strings, the way to document an optional parameter is `ref_count (int, optional): ...`... [see here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

Reply:
> Don't use **kwargs for everything; use explicit arguments.

I'll take this request as proper style in Spack, and will implement it. But if we are talking about idiomatic python, I think the subject is [at least controversial](https://stackoverflow.com/a/1098834/771663).

Reply:
Yeah I agree it’s controversial. I like it because it makes the prototypes more readable and easier to document. It’s also closer to the way you’d write it in Python 3, where you can get the best of both worlds and force kwargs. I’d argue that doing things more like 3 is the better way to go... maybe one day we can drop 2 support altogether... one day far away 😭.

Reply:
:+1: for dropping py2 👼

Reply:
There will probably be hpcpythonclock.org or craypythonclock.org or something long after pythonclock.org has counted down 😭